Designated Initializers don't work as expected

Hi,

I tried to use the Designated Initializers feature, but it doesn't seem work as (I) expected. For example, this is valid C++20:

```
class A {
  public:
    int a;
    int b;
};

class B {
  public:
    B(int x, int y);

  private:
    A foo;
};

B::B(int x, int y) {
    this->foo = {
        .b = y
    };
}
```

However, this doesn't:

```
#define VULKAN_HPP_NO_STRUCT_CONSTRUCTORS
#include <vulkan/vulkan.hpp>

class MyVulkanTest
{
  public:
    MyVulkanTest();

  private:
    vk::ApplicationInfo applicationInfo;
};

MyVulkanTest::MyVulkanTest()
{
    this->applicationInfo = {
        .pApplicationName   = "My Application",
        .applicationVersion = VK_MAKE_VERSION(0,0,1),
        .pEngineName        = "My Engine",
        .engineVersion      = VK_MAKE_VERSION(0,0,1),
        .apiVersion         = VK_API_VERSION_1_0
    };
}
```

Both Clang (11) and GCC 9/10 throw the same error:

```
clang++ -Og -pipe -std=c++20 -c -o MyVulkanTest.o MyVulkanTest.cpp
MyVulkanTest.cpp:15:27: error: no viable overloaded '='
    this->applicationInfo = {
    ~~~~~~~~~~~~~~~~~~~~~ ^ ~
/usr/include/vulkan/vulkan.hpp:18094:23: note: candidate function not viable: cannot convert initializer list argument to 'const VkApplicationInfo'
    ApplicationInfo & operator=( VkApplicationInfo const & rhs ) VULKAN_HPP_NOEXCEPT
                      ^
/usr/include/vulkan/vulkan.hpp:18100:23: note: candidate function not viable: cannot convert initializer list argument to 'const vk::ApplicationInfo'
    ApplicationInfo & operator=( ApplicationInfo const & rhs ) VULKAN_HPP_NOEXCEPT
                      ^
1 error generated.
make: *** [<builtin>: MyVulkanTest.o] Error 1
```

(The one from GCC is the same but worded a bit differently.)

After some mucking around I found that as soon as I introduce a private member in the first example (`a` and `b` can still remain `public`) this example throws the same error as the Vulkan example. Apparently Designated Initializers work *only* for classes that have *no* private or protected (static) members.

So my question is: Is it just me that misunderstood how to use Designated Initializers? Or is this a limitation in `vulkan.hpp` that may be fixed in a later version?

Best regards
Philipp

--- a/tests/DesignatedInitializers/DesignatedInitializers.cpp
+++ b/tests/DesignatedInitializers/DesignatedInitializers.cpp
@@ -21,9 +21,32 @@
 
 #include <iostream>
 
-#if defined(__clang__) || defined(__GNUC__)
-#pragma GCC diagnostic ignored "-Wunused-variable"
+#if defined( __clang__ ) || defined( __GNUC__ )
+#  pragma GCC diagnostic ignored "-Wunused-variable"
 #endif
+
+class MyVulkanTest
+{
+public:
+  MyVulkanTest();
+
+private:
+  vk::ApplicationInfo applicationInfo;
+};
+
+MyVulkanTest::MyVulkanTest()
+  : applicationInfo{ .pApplicationName   = "My Application",
+                     .applicationVersion = VK_MAKE_VERSION( 0, 0, 1 ),
+                     .pEngineName        = "My Engine",
+                     .engineVersion      = VK_MAKE_VERSION( 0, 0, 1 ),
+                     .apiVersion         = VK_API_VERSION_1_0 }
+{
+  this->applicationInfo = vk::ApplicationInfo{ .pApplicationName   = "My Application",
+                                               .applicationVersion = VK_MAKE_VERSION( 0, 0, 1 ),
+                                               .pEngineName        = "My Engine",
+                                               .engineVersion      = VK_MAKE_VERSION( 0, 0, 1 ),
+                                               .apiVersion         = VK_API_VERSION_1_0 };
+}
 
 int main( int /*argc*/, char ** /*argv*/ )
 {

